packages/e2e-test-utils/src/transform-block-to.js

```diff
@@ -7,17 +7,10 @@ export async function transformBlockTo( name ) {
 	await page.mouse.move( 200, 300, { steps: 10 } );
 	await page.mouse.move( 250, 350, { steps: 10 } );
 	await page.click( '.block-editor-block-switcher__toggle' );
-	// Close the "Block Styles" section if it is open.
-	// Having the section open may make the transform buttons hidden on the testing resolution.
-	const closeBlockStylesButton = await page.$x(
-		'//div[contains(@class,"block-editor-block-switcher__popover")]//button[contains(text(),"Block Styles")][@aria-expanded="true"]'
-	);
-	if ( closeBlockStylesButton.length > 0 ) {
-		await closeBlockStylesButton[ 0 ].click();
-	}
 	const insertButton = ( await page.$x(
 		`//button//span[contains(text(), '${ name }')]`
 	) )[ 0 ];
+	await insertButton.evaluate( ( element ) => element.scrollIntoView() );
```

Comment:
It's a bit surprising to me that it's necessary, given that the default implementation of `click` should be doing the scrolling already on our behalf:

https://github.com/puppeteer/puppeteer/blob/d17708ba1fc8411941bfd07f633aaf2fb5cb671e/lib/JSHandle.js#L278-L285

I wonder if it has to do with the fact that the element is in it's own scrollable region, not the page's default scroll area. If that's the case, seems like an area of improvement for Puppeteer to consider.

Reply:
Yeah, I stumbled upon the same code and the same issues. It's not exactly the same as `element.scrollIntoView()` will force the element to the top even if it's already in view.